lib/checkother.cpp

```diff
@@ -2690,7 +2690,8 @@ void CheckOther::checkEvaluationOrder()
                         continue; // don't care about sizeof usage
                     tokens.push(tok3->astOperand1());
                     tokens.push(tok3->astOperand2());
-                    if (isSameExpression(mTokenizer->isCPP(), false, tok->astOperand1(), tok3, mSettings->library, true)) {
+                    if (isSameExpression(mTokenizer->isCPP(), false, tok->astOperand1(), tok3, mSettings->library, true) && 
```

Comment:
I am guessing that it's better that followVariables is disabled unless it's explicitly enabled. How about an extra argument for `isSameExpression` that is `bool followVariables = false`? Then we can explicitly set it to true where we think it makes sense.

Reply:
> How about an extra argument for isSameExpression that is bool followVariables = false?

I can definitely add that. Although in this case would we still want to warn when the two expressions are `x+y` and `y+x`? If this is the case, then we most likely dont want to use `expressionString` as an extra check.

If this is not the case, then we would still want the `expressionString` check, and the `followVariables = false` can help improve the performance of `isSameExpression` in this context.

Reply:
> I can definitely add that. Although in this case would we still want to warn when the two expressions are x+y and y+x? If this is the case, then we most likely dont want to use expressionString as an extra check.

In my humble opinion we do want to warn for `x+y` and `y+x` in this check. I guess something like this should generate an error:

    a[x+y] = a[x+y]++;

So please remove that extra `expressionString` check.